include/dca/function/function.hpp

```diff
@@ -47,7 +50,9 @@ class function {
   // Default constructor
   // Constructs the function with the name name.
   // Postcondition: All elements are set to zero.
-  function(const std::string& name = default_name_);
+  // Special case: when distributed_g4_enabled, G4 related variables only gets
+  // allocation of 1/p of original G4 size, where p = #mpiranks
+  function(const std::string& name = default_name_, const bool distributed_g4_enabled = false);
```

Comment:
This constructor should have a unit test to insure the expected allocation continues to be made in the future. As well as to guard the default argument value.

Reply:
Not sure if I understand it, but what does it mean "the expected allocation continues to be made in the future."? Can you explain it a bit more? So I can better prepare a unit test for it. Thanks.

Reply:
I'm going to be working in function soon so I will explain address it then. This is fine for this PR.